src/IndexingMsg.chpl

```diff
@@ -13,9 +13,136 @@ module IndexingMsg
 
     use CommAggregation;
 
+    use FileIO;
+    use List;
+
     private config const logLevel = ServerConfig.logLevel;
     const imLogger = new Logger(logLevel);
 
+    proc jsonToTuple(json: string, type t) throws {
+        var f = opentmp(); defer { ensureClose(f); }
+        var w = f.writer();
+        w.write(json);
+        w.close();
+        var r = f.reader(start=0);
+        var tup: t;
+        r.readf("%jt", tup);
+        r.close();
+        return tup;
+    }
+
+    proc arrayViewMixedIndexMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
+        var (pdaName, indexDimName, ndimStr, dimProdName, coords) = payload.splitMsgToTuple(5);
+        var ndim = try! ndimStr:int;
+        imLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                                    "%s %s %i %s %s".format(cmd, pdaName, ndim, dimProdName, coords));
+
+        var dimProd: borrowed GenSymEntry = getGenericTypedArrayEntry(dimProdName, st);
+        var dimProdEntry = toSymEntry(dimProd, int);
+
+        var indexDim: borrowed GenSymEntry = getGenericTypedArrayEntry(indexDimName, st);
+        var indexDimEntry = toSymEntry(indexDim, int);
+        ref dims = indexDimEntry.a;
+
+        // String array containing the type of the following value at even indicies then the value ex. ["int", "7", "slice", "(0,5,-1)", "pdarray", "id_4"]
+        var typeCoords: [0..#(ndim*2)] string = jsonToPdArray(coords, ndim*2);
+
+        var scaledCoords: [makeDistDom(+ reduce dims)] int;
+        // check there's enough room to create a copy for scan and throw if creating a copy would go over memory limit
+        overMemLimit(numBytes(int) * dims.size);
+        var offsets = (+ scan dims) - dims;
+
+        for i in 0..#typeCoords.size by 2 {
+            select typeCoords[i] {
+                when "int" {
+                    scaledCoords[offsets[i/2]] = typeCoords[i+1]:int * dimProdEntry.a[i/2];
+                }
+                when "slice" {
+                    var (start, stop, stride) = jsonToTuple(typeCoords[i+1], 3*int);
+                    var slice: range(stridable=true) = convertSlice(start, stop, stride);
+                    var scaled: [0..#slice.size] int = slice * dimProdEntry.a[i/2];
+                    for j in 0..#slice.size {
+                        scaledCoords[offsets[i/2]+j] = scaled[j];
+                    }
+                }
+                // Advanced indexing not yet supported
+                // when "pdarray" {
```

Comment:
Were you seeing issues running this code?

Reply:
ah yeah i should make a note of this. This code was working the way I intended but numpy doesn't handle array indexing the way I expected. This is talked about a bit in https://numpy.org/doc/stable/user/basics.indexing.html

but I expected slice and pdarray to behave the same when their indices matched up, but that's not always the case
```python
>>> n = np.arange(4).reshape(2,2)
# sometimes they line up
>>> n[:,:]
array([[0, 1],
       [2, 3]])

>>> n[:,[0,1]]
array([[0, 1],
       [2, 3]])

>>> n[[0,1],:]
array([[0, 1],
       [2, 3]])
# sometimes they do not
>>> n[[0,1],[0,1]]
array([0, 3])
```
basically when you have 2+ arrays, the functionality switches from the Cartesian product of coordinates to more coordinate-wise.

so `n[:, :]` gets `[0,0], [0,1], [1,0], [1,1]` whereas `n[[0,1],[0,1]]` only gets `[0,0], [1,1]`

Reply:
We should make sure to document this when documentation is added. At least link to the numpy link to indicate this is the expected functionality.